### PR TITLE
:seedling: Add powerOffMode to templates

### PR DESCRIPTION
**Pull request by laozc**
**What this PR does / why we need it**:
Adds `powerOffMode` to the current templates to serve as documentation to control the VM power off behavior.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: templates/cluster-template-external-loadbalancer.yaml [hunk 1]
```diff
@@ -60,6 +60,7 @@ spec:
       storagePolicyName: '${VSPHERE_STORAGE_POLICY}'
       template: '${VSPHERE_TEMPLATE}'
       thumbprint: '${VSPHERE_TLS_THUMBPRINT}'
```
Comment (sbueringer) [inline templates/cluster-template-external-loadbalancer.yaml:62]: I would have to look up the details but I think we have to modify flavorgen and then re-generate those files

Comment (sbueringer) [reply templates/cluster-template-external-loadbalancer.yaml:62]: Related: https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/issues/2153

**Review by zhanggbj — COMMENTED** [review]
@laozc it seems cluster-template-node-ipam.yaml also have `VSphereMachineTemplate` but not specified with powerOffMode, is this by intention?

**Comment by laozc** [comment]
> @laozc it seems cluster-template-node-ipam.yaml also have `VSphereMachineTemplate` but not specified with powerOffMode, is this by intention?

Thanks for catching it.
Also set the powerOffMode explicitly.

**Comment by sbueringer** [comment]
Thank you!!

/lgtm
/approve

**Pull request merged**